src/dashboard/highlightedMatch/matchData.ts

```diff
@@ -0,0 +1,123 @@
+import { addDots, addSelector, clearSelectors, setImportStatus } from '../globalScripts';
+import { ImportStatus } from 'types/importStatus';
+import { HighlightedMatch, NextTeams, TournamentData } from 'schemas';
+import { Match } from './Match';
+
+const highlightedMatchData = nodecg.Replicant<HighlightedMatch>('highlighedMatches');
+const tournamentData = nodecg.Replicant<TournamentData>('tournamentData');
+const nextTeams = nodecg.Replicant<NextTeams>('nextTeams');
+
+const matchDataStatusElem = document.getElementById('match-data-status');
+const stageSelectElem = document.getElementById('stage-selector') as HTMLSelectElement;
+const matchSelectElem = document.getElementById('match-selector') as HTMLSelectElement;
+const getMatchesBtnElem = document.getElementById('get-matches') as HTMLButtonElement;
+const setNextMatchBtnElem = document.getElementById('set-next-match-btn') as HTMLButtonElement;
+
+const teamAName = document.getElementById('team-a-name');
+const teamBName = document.getElementById('team-b-name');
+
+/**
+ * Gets match from matchID
+ * @param id the ID of the match
+ */
+function getMatchFromID(id: string): Match | void {
+    for (let x = 0; x < highlightedMatchData.value.length; x++) {
+        const value = highlightedMatchData.value[x];
+        if (value.meta.id === id) {
+            return value;
+        }
+    }
+}
+
+// When get match button is pressed
+document.getElementById('get-matches').onclick = () => {
```

Comment:
I won't make you remove these comments, but the things some of them reference feel very self-explanatory.